postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala

```diff
@@ -35,6 +38,59 @@ object FunctionDefSpec extends PostgresRunnableSpec with ShopSchema {
         r <- testResult.runCollect
       } yield assert(r.head)(equalTo(expected))
 
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
+    },
+    testM("make_date") {
+      val query = select(MakeDate(2013, 7, 15)) from customers
+
+      val expected = LocalDate.of(2013, 7, 15)
+
+      val testResult = execute(query).to[LocalDate, LocalDate](identity)
+
+      val assertion = for {
+        r <- testResult.runCollect
+      } yield assert(r.head)(equalTo(expected))
+
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
+    },
+//    testM("make_interval") {
+//      //make_interval(days => 10) → 10 days
+//      ???
+//    },
+    testM("make_time") {
+      val query      = select(MakeTime(8, 15, 23.5)) from customers
+      val expected   = LocalTime.parse("08:15:23.5")
+      val testResult = execute(query).to[LocalTime, LocalTime](identity)
+
+      val assertion = for {
+        r <- testResult.runCollect
+      } yield assert(r.head)(equalTo(expected))
+
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
+    },
+    testM("make_timestamp") {
```

Comment:
Both this and the make_timestampz complain that there are two many arguments, I'm not really sure what's going on there. Also not really sure how to tackle the case of interval which takes a format of "<interval> => <value>) Maybe a custom parser for that?

Reply:
RE the timestamps, in ExprModule.Expr there are FunctionCall1 through 4 and corresponding overloaded apply methods on FunctionDef. You've got 6 arguments here, so you need a FunctionCall6 and an apply method on FunctionDef that can make one. Bonus points for 5 while you're at it ;). As a side note, I don't think the options are necessary here as I think they're all required. If not you could just use default values (0) for all

for make interval, you could use default values for all the arguments which would give a nice API. I think you're up for FunctionCall7 as well...

Reply:
@robmwalsh thanks so much for the help with this! I have to step away for the evening but will try and get this wrapped up by tomorrow!